posthog/models/feature_flag/local_evaluation.py

```diff
@@ -0,0 +1,174 @@
+    else "replica"
+)
+
+flags_hypercache = HyperCache(
+    namespace="feature_flags",
+    value="flags_with_cohorts.json",
+    load_fn=lambda key: _get_flags_response_for_local_evaluation(HyperCache.team_from_key(key), include_cohorts=True),
+)
+
+flags_without_cohorts_hypercache = HyperCache(
+    namespace="feature_flags",
+    value="flags_without_cohorts.json",
+    load_fn=lambda key: _get_flags_response_for_local_evaluation(HyperCache.team_from_key(key), include_cohorts=False),
+)
+
+
+def get_flags_response_for_local_evaluation(team: Team, include_cohorts: bool) -> dict | None:
+    return (
+        flags_hypercache.get_from_cache(team)
+        if include_cohorts
+        else flags_without_cohorts_hypercache.get_from_cache(team)
+    )
+
+
+def update_flag_caches(team: Team):
+    flags_hypercache.update_cache(team)
+    flags_without_cohorts_hypercache.update_cache(team)
+
+
+def clear_flag_caches(team: Team, kinds: list[str] | None = None):
+    flags_hypercache.clear_cache(team, kinds=kinds)
+    flags_without_cohorts_hypercache.clear_cache(team, kinds=kinds)
+
+
+def _get_flags_for_local_evaluation(team: Team):
+    feature_flags = FeatureFlag.objects.db_manager(DATABASE_FOR_LOCAL_EVALUATION).filter(
+        ~Q(is_remote_configuration=True),
+        team__project_id=team.project_id,
+        deleted=False,
+    )
+
+    return feature_flags
+
+
+def _get_flags_with_cohorts_for_local_evaluation(team: Team) -> tuple[list[FeatureFlag], dict]:
+    feature_flags = _get_flags_for_local_evaluation(team)
+
+    cohorts = {}
+    seen_cohorts_cache: dict[int, CohortOrEmpty] = {}
+
+    try:
+        seen_cohorts_cache = {
+            cohort.pk: cohort
+            for cohort in Cohort.objects.db_manager(DATABASE_FOR_LOCAL_EVALUATION).filter(
+                team__project_id=team.project_id, deleted=False
+            )
+        }
+    except Exception:
+        logger.error("Error prefetching cohorts", exc_info=True)
+
+    for feature_flag in feature_flags:
+        try:
+            filters = feature_flag.get_filters()
+            # transform cohort filters to be evaluated locally
+            if (
+                len(
+                    feature_flag.get_cohort_ids(
+                        using_database=DATABASE_FOR_LOCAL_EVALUATION,
+                        seen_cohorts_cache=seen_cohorts_cache,
+                    )
+                )
+                == 1
+            ):
+                feature_flag.filters = {
+                    **filters,
+                    "groups": feature_flag.transform_cohort_filters_for_easy_evaluation(
+                        using_database=DATABASE_FOR_LOCAL_EVALUATION,
+                        seen_cohorts_cache=seen_cohorts_cache,
+                    ),
+                }
+            else:
+                feature_flag.filters = filters
+
+            cohort_ids = feature_flag.get_cohort_ids(
```

Comment:
When I tested this, the `cohorts` section was empty when I added `send_cohorts&use_cache` to the query string.

The cohort_ids are captured after the flag transformation, but the transformation removes cohort references. This causes the cohorts dict to be empty even when flags use cohorts.

Fix: Move `cohort_ids = feature_flag.get_cohort_ids(...)` to before the transformation (around line 81).

Reply:
Ok, I pushed my fix.